### DISCUSSION TITLE
Media Init Failed if SD memory card is not formatted in fat32

### DISCUSSION THREAD

**Discussion by bailed00** [mriscoc/Ender3V2S1#680] [Q&A] [answered]
EDIT: Solved by https://github.com/mriscoc/Ender3V2S1/discussions/680#discussioncomment-4781244

After installing [Ender3V2-422-BLTUBL-T13-MPC-20221222.bin] everything is great except my microSD reader will not read any card I insert (media init failed) on both of my printers. I had the previous version working awesomely

**Comment by jxer** [comment]
I have new issues with sd card as well. I have one printer going into endless reboot, until sd is removed , once it boots up its fine, I had 2 others crash during a sd card read as well.

**Comment by bailed00** [comment, marked as answer]
I apologize for my not being more thorough in solving my own issue. It
appears that I had mixed in some exfat formatted cards. It definitely did
not like that so I reformatted to fat32 and the problem was solved. It
might be worth checking

**Comment by jxer** [comment]
my sd cards need leave my machine. I also reformatted and moved cards. still is occuring

**Comment by jxer** [comment]
but its not a media init fail, the printer doesn't load mesh, and goes into endless reboot, until the sd card is removed.

**Comment by Amatsugu** [comment]
> I have new issues with sd card as well. I have one printer going into endless reboot, until sd is removed , once it boots up its fine, I had 2 others crash during a sd card read as well.

I have the exact same issue, Endless boot loop until sd card is removed. Sdcard is FAT32

**Comment by GraemeDBlue** [comment]
I am getting the same , only happened after updating this firmware, I can see the following in the Octoprint terminal logs if this is helpful?

```Send: N0 M110 N0*125
Recv: start
Printer sent 'start' while printing. External reset? Aborting job since printer lost state.
Send: N0 M110 N0*125
Changing monitoring state from "Starting" to "Cancelling"
Recv:  Watchdog Reset
Recv: Marlin ProUI 2.1.3 MRiscoC
Recv: Build: 20221222 - 02:12
Recv: Author: Miguel A. Risco-Castillo (MRiscoC)
Recv: echo: Free Memory: 37015  PlannerBufferBytes: 1856
Recv: echo:V87 stored settings retrieved (760 bytes; crc 14128)
Recv: //action:notification Stored settings retrieved
Recv: Unified Bed Leveling System v1.01 inactive
Recv: Mesh loaded from slot 0
Recv: Mesh 0 loaded from storage.
Recv: //action:prompt_end
Recv: BL24CXX Check succeeded!```

**Comment by codetak** [comment]
I have the same issue, I've been using the same SDCard without issues until the latest update. Endless boot loop until sd is removed. If "Media Update" is enabled, printer reboots when sd is inserted.

My configuration is: **Ender3V2-422-BLTUBL-HomeOffs-IS-LA**
Workaround: Media Update=disabled, start printer without SD, insert SD

**Comment by mriscoc** [comment]
> Recv:  Watchdog Reset

The kernel is waiting for an answer from the SD card but it is not responding, so a Watchdog Reset is issued.

**Comment by codetak** [comment]
> > Recv:  Watchdog Reset
> 
> The kernel is waiting for an answer from the SD card but it is not responding, so a Watchdog Reset is issued.

It seems that randomly, after 6-7 reboots it manages to boot. Serial log bellow:

start
 Watchdog Reset
Marlin ProUI 2.1.3 MRiscoC
Build: 20221228 - 13:20
Author: Miguel A. Risco-Castillo (MRiscoC)
echo: Free Memory: 33911  PlannerBufferBytes: 1792
echo:V87 stored settings retrieved (754 bytes; crc 33932)
//action:notification Stored settings retrieved
Unified Bed Leveling System v1.01 inactive
Mesh loaded from slot 0
Mesh 0 loaded from storage.
//action:prompt_end
BL24CXX Check succeeded!
start
 Watchdog Reset
Marlin ProUI 2.1.3 MRiscoC
Build: 20221228 - 13:20
Author: Miguel A. Risco-Castillo (MRiscoC)
echo: Free Memory: 33911  PlannerBufferBytes: 1792
echo:V87 stored settings retrieved (754 bytes; crc 33932)
//action:notification Stored settings retrieved
Unified Bed Leveling System v1.01 inactive
Mesh loaded from slot 0
Mesh 0 loaded from storage.
//action:prompt_end
BL24CXX Check succeeded!
start
 Watchdog Reset
Marlin ProUI 2.1.3 MRiscoC
Build: 20221228 - 13:20
Author: Miguel A. Risco-Castillo (MRiscoC)
echo: Free Memory: 33911  PlannerBufferBytes: 1792
echo:V87 stored settings retrieved (754 bytes; crc 33932)
//action:notification Stored settings retrieved
Unified Bed Leveling System v1.01 inactive
Mesh loaded from slot 0
Mesh 0 loaded from storage.
//action:prompt_end
BL24CXX Check succeeded!
start
 Watchdog Reset
Marlin ProUI 2.1.3 MRiscoC
Build: 20221228 - 13:20
Author: Miguel A. Risco-Castillo (MRiscoC)
echo: Free Memory: 33911  PlannerBufferBytes: 1792
echo:V87 stored settings retrieved (754 bytes; crc 33932)
//action:notification Stored settings retrieved
Unified Bed Leveling System v1.01 inactive
Mesh loaded from slot 0
Mesh 0 loaded from storage.
//action:prompt_end
BL24CXX Check succeeded!
start
 Watchdog Reset
Marlin ProUI 2.1.3 MRiscoC
Build: 20221228 - 13:20
Author: Miguel A. Risco-Castillo (MRiscoC)
echo: Free Memory: 33911  PlannerBufferBytes: 1792
echo:V87 stored settings retrieved (754 bytes; crc 33932)
//action:notification Stored settings retrieved
Unified Bed Leveling System v1.01 inactive
Mesh loaded from slot 0
Mesh 0 loaded from storage.
//action:prompt_end
BL24CXX Check succeeded!
start
 Watchdog Reset
Marlin ProUI 2.1.3 MRiscoC
Build: 20221228 - 13:20
Author: Miguel A. Risco-Castillo (MRiscoC)
echo: Free Memory: 33911  PlannerBufferBytes: 1792
echo:V87 stored settings retrieved (754 bytes; crc 33932)
//action:notification Stored settings retrieved
Unified Bed Leveling System v1.01 inactive
Mesh loaded from slot 0
Mesh 0 loaded from storage.
//action:prompt_end
BL24CXX Check succeeded!
echo:SD card ok
�echo: Serial 0 baud rate set to 250000
ok P15 B16

**Comment by lookalek** [comment]
Hi guys,

I am also having issues with my SD card. Same as described above, sometimes after inserting the card, my printer falls into reboot loop until the card is removed. After a few trials, the printer finally reads the card correctly.

Also, sometimes inserting the card doesn't cause the reboot immediately, but it starts after entering the media folder on the main screen of the printer. Switching the printer on and off with a power switch also helps.

The card is formatted to FAT32. Recreating its partitions doesn't help either, same story.

**Comment by BigApple13** [comment]
I was able to fix it by formating the sd card using windows. I used FAT32 and allocation size unit set to default.

**Comment by lookalek** [comment]
> I was able to fix it by formating the sd card using windows. I used FAT32 and allocation size unit set to default.

Tried that many times, helps for a while only in my case.
Even used CMD's DiskPart to wipe the card completely and create a new FAT32 partition.

**Comment by jxer** [comment]
So an oddity, my extruder fan was slowing failing, and the printer would reboot constantly with the sd card in. I changed the fan this weekend, and the issue has gone away. It obviously was some kind of emi rfi, issue, but ultimately, may be a timing issue on sd card loading. something definitely changed. changing the fan was an interesting fix.

**Comment by SorenRR** [comment]
Try using Rufus to format your SD card ... here; [https://github.com/pbatard/rufus/releases](https://github.com/pbatard/rufus/releases)

I used Rufus to format a 64 GB PNY Elite Category 10 card as FAT32. My Windows 10 only offered exFAT !

**Comment by jxer** [comment]
> Try using Rufus to format your SD card ... here; https://github.com/pbatard/rufus/releases
> 
> I used Rufus to format a 64 GB PNY Elite Category 10 card as FAT32. My Windows 10 only offered exFAT !

its not a card formatting issue, really need to stop saying this, my sd cards never change and I have 5 printers, only 1 was doing it.

**Comment by SorenRR** [comment]
> > Try using Rufus to format your SD card ... here; https://github.com/pbatard/rufus/releases
> > I used Rufus to format a 64 GB PNY Elite Category 10 card as FAT32. My Windows 10 only offered exFAT !
> 
> its not a card formatting issue, really need to stop saying this, my sd cards never change and I have 5 printers, only 1 was doing it.

I'm using Octoprint now so I wouldn't know - except that Rufus can do miracles for a cheap SD Card.

**Comment by RacerEXE** [comment]
For me reading the card fails unless I have "media update" enabled. When I enable the checkbox, it works fine.

**Comment by mriscoc** [comment]
For general Q&A please post in https://github.com/mriscoc/Ender3V2S1/discussions

**Reply by codetak** [reply]
The issue is definitely there @mriscoc . Everybody is trying to provide you more information about the behaviour of the firmware. This is a bug, not a discussion. I have also checked to moving back to an older release and the issue was gone. I restored back to the latest and issue re-appeared. btw, I don't use the precompiled binaries if that's somehow related, I compile in Vscode.

**Comment by lookalek** [comment]
> For general Q&A please post in https://github.com/mriscoc/Ender3V2S1/discussions

We are not having general discussion here but trying to describe the issue while eliminating external factors.
Issue is there, and is definitely not solved.

**Comment by mriscoc** [comment]
Thousands of people use the precompiled firmware without SD card issues, so, the current firmware version doesn't have general problems with the SD cards. If you read all the thread, some people have problems with card formatting, bad cards, bad slot, or with their own custom compile with unknown changes that maybe broke something, so that is outside of the free support that we do here.

Because it is not a firmware issue, I could simply close it but I prefer to move this conversation to Discussion to allow other people with the same problem to share his own solution for the benefit of others.

**Reply by codetak** [reply]
We all appreciate you work @mriscoc and we are really thankful. We really don't expect more support, you've done more than enough just by providing your fw as is. But just in matters of resolving this and helping everyone, how do you explain this happening only in the current version for everyone and been resolved by restoring to the prior version? I checked using one of your pre-compiled binaries as well, I get the same issue.

**Reply by mriscoc** [reply]
As of build 20220816, the SD card file system is not re-read each time the media menu is entered unless the `Media update` option is checked in the Advanced settings menu, not more changes were made to SD routines after that. But maybe there are changes in Marlin core, you could read the PR's merged in the Marlin repository.

**Comment by jxer** [comment]
i only run precompiled firmwares, and have 5 printers, something has changed, so lets not pretend its some kind of sd card issue or slot issue, these printers don't move, and the sd cards don't move, they are not even ejected generally. when 3 printers had hiccups, that's more than a coincidence. something changed.

**Reply by mriscoc** [reply]
Does your printer have SD media extenders?

**Reply by jxer** [reply]
No

**Comment by codetak** [comment]
@mriscoc this seems to be a known issue that in earlier versions did not affect Creality boards:
https://github.com/MarlinFirmware/Marlin/issues/21524
Thus, this is not a matter of SD formatting or bad SD. This is a bug for sure.

**Reply by mriscoc** [reply]
> MarlinFirmware/Marlin#21524

According to that report, it seems that Marlin accesses the SD card in a faster DMA way (https://github.com/MarlinFirmware/Marlin/pull/21476) that before, which could cause problems with low-quality or slow SD cards. So, for your particular case you need to revert that commit. Since this only affects a very small number of users while benefiting many more, it is expected that no fix will be provided in Marlin.

**Reply by codetak** [reply]
This is not the case though, I don’t have the problem with untouched Marlin. Only in your custom fw. Those commits date back to 2 years ago.

**Reply by mriscoc** [reply]
Ok thank you. One thing more, be sure that there is no PLR file inside of your memory card.

**Reply by codetak** [reply]
You mean Power-Loss-Recovery file? If so, no. I have tried formatting the card and problem persists. If this is somehow related, I'm on 4.2.2 board.

**Comment by mriscoc** [comment]
Informative link: https://www.3dprintgorilla.com/ender-3-pro-v2-neo-s1-sd-card-size
